endpoints-framework-tools/src/main/java/com/google/api/server/spi/tools/EndpointsToolAction.java

```diff
@@ -359,6 +366,23 @@ protected Option makeBasePathOption() {
     return urls.build().toArray(new URL[0]);
   }
 
+  protected List<String> getServiceClassNames(String warPath) {
+    if (!getArgs().isEmpty()) {
+      return getArgs();
+    }
+    try {
+      File warDir = new File(warPath).getAbsoluteFile();
+      File webInfDir = new File(warDir, "WEB-INF");
+      File webXmlFile = new File(webInfDir, "web.xml");
+      if (webXmlFile.exists()) {
+        return WebXml.parse(webXmlFile).endpointsServiceClasses();
+      }
+    } catch (ParserConfigurationException | IOException | SAXException e) {
```

Comment:
I think the log is sufficient. If it reaches this point, the tool will error out anyway since no classes have been specified. It's not really our job to validate the web.xml.